junit-jupiter-api/src/main/java/org/junit/jupiter/api/Assertions.java

```diff
@@ -54,6 +54,9 @@ private Assertions() {
 	 * <p>See Javadoc for {@link #fail(String, Throwable)} for an explanation of
 	 * this method's generic return type {@code V}.
 	 */
+	// we purposely return a V that has nothing to do with the parameters of this method; see
+	// the javadoc for details.
+	@SuppressWarnings("TypeParameterUnusedInFormals")
```

Comment:
@sbrannen We have two options AFAICT:
1. Accept the `@SuppressWarnings("TypeParameterUnusedInFormals")`, at the expense of your sanity. :wink:
2. Globally disable the rule.

I'm personally not a fan of disabling the rule, because according the [bug patterns docs](http://errorprone.info/bugpatterns) (type Ctrl+F and search for "TypeParameterUnusedInFormals"), it's a default non-experimental rule, which suggests to me that it's worth keeping on.

But on the other hand, I empathise with you on annoying warnings in IDEs, so if it really does drive you round the bend, then I'd be happy to disable it.